technical-architecture.md

```diff
@@ -0,0 +1,84 @@
+
+## <a id="overview"></a>Overview
+Application Service Adapter implements a subset of the [v3 Cloud Foundry APIs](https://v3-apidocs.cloudfoundry.org/) in order to support common Cloud Foundry developer workflows. Application Service Adapter is installed directly on to a Kubernetes cluster than has Tanzu Application Platform (TAP) installed and provides a CF API translation layer that converts CF API calls into underlying and Kubernetes resources. In addition to this API, Application Service Adapter also provides a set of Kubernetes [custom resources](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/), [controllers](https://kubernetes.io/docs/concepts/architecture/controller/), and [admission webhooks](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/).
+
+## <a id="high-level-architecture"></a>High-level architecture
+
+![High-level Application Service Adapter architecture diagram](images/tas-adapter-high-level-architecture.png)
+
+### Components
+Application Service Adapter includes components from the open source Cloud Foundry Korifi project as well as a number of proprietary components for deeper integration with Tanzu Application Platform.
+
+The Korifi components are primarily tasked with implementing the Cloud Foundry data model and APIs. They are:
+
+* **Korifi CRDs:** A set of Kubernetes custom resources under the korifi.cloudfoundry.org Api Group that implement the core set of V3 Cloud Foundry resources. These custom resources replicate the Cloud Foundry data model and are used to store persistent state for the CF API translation layer. These resources can be interacted with directly by existing Kubernetes tooling (e.g. the kubectl CLI) and serve as an extension point for further interoperability.
+* **Korifi API Deployment:** A Golang implementation of a core set of V3 Cloud Foundry APIs that is backed by the Korifi CRDs. Existing Cloud Foundry API clients (such as the cf CLI) can target the Korifi API and continue to use their existing CF developer workflows.
+* **Korifi Controllers Deployment:** A set of Kubernetes controllers that implement CF subsystems by orchestrating and reconciling the Korifi CRDs into consolidated intermediate resources that contain the information necessary to build and run an application. This deployment also runs admission webhooks that validate and normalize Korifi resources.
+
+Application Service Adapter converts these intermediate resources using its default builder/runner controller implementations into TAP and Kubernetes resources such as Tanzu Build Service Images for building applications and Kubernetes StatefulSets for running them.
+
+## <a id="cartographer-integration"></a>Experimental Cartographer integration
+
+Application Service Adapter includes an (optional) experimental builder and runner implementation that utilizes TAP's Cartographer Supply Chains to manage the build and run stages of the application lifecycle.
```

Comment:
@matt-royal I didn't want to go too in-depth on this part since it is a rapidly evolving part of the product, but this is an area we should flesh out more in the future.